src/mdns/mdns_mdnssd.cpp

```diff
@@ -1056,7 +1043,7 @@ otbrError PublisherMDnsSd::ServiceInstanceResolution::GetAddrInfo(uint32_t aInte
 
     otbrLogInfo("DNSServiceGetAddrInfo %s inf %d", mInstanceInfo.mHostName.c_str(), aInterfaceIndex);
 
-    dnsError = DNSServiceGetAddrInfo(&mServiceRef, kDNSServiceFlagsTimeout, aInterfaceIndex,
+    dnsError = DNSServiceGetAddrInfo(&mServiceRef, /* flags */ 0, aInterfaceIndex,
```

Comment:
Looks like the subscription will not stop now? Should we pass the mServiceRef to DNSServiceRefDeallocate() when unsubscribing service?

Reply:
When unsubscribing a service, the `ServiceInstanceResolution` will be destructed and its base class `ServiceRef` will be destructed. The destructor of `ServiceRef` calls `DNSServiceRefDeallocate()`.

Reply:
I see, then the PublisherMDnsSd::UnsubscribeService with empty aInstanceName does seem related to this PR, as it is not destructing the SubscribedServices thus leaking the ServiceRefs?

Reply:
Explained via offline chat. The short answer is, the `DNSServiceRefDeallocate()` is triggered on the destruction of relevant objects like `ServiceSubscription` and `ServiceInstanceResolution`.